cgi-bin/DW/Controller/API/REST/Icons.pm

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/perl
+#
+# DW::Controller::API::REST::Icons
+#
+# API controls for the icon system
+#
+# Authors:
+#      Allen Petersen <allen@suberic.net>
+#
+# Copyright (c) 2016 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::Controller::API::Rest::Icons;
+use DW::Controller::API::REST;
+use strict;
+use warnings;
+use DW::Routing;
+use DW::Request;
+use DW::Controller;
+use JSON;
+
+my $icons_all = DW::Controller::API::REST->path('icons_all.yaml', 1, {'get' => \&rest_get});
+
+sub rest_get {
+    my ( $self, $opts, $username, $picid ) = @_;
+
+    my $u = LJ::load_user( $username );
+
+    # we want to handle the not logged in case ourselves
+    my ( $ok, $rv ) = controller( anonymous => 1 );
```

Comment:
move this above the user load, just in case it returns error and we bail -- might as well not load a user in that case

Reply:
or just don't need it at all :)